### PR TITLE
Added support for landscape view

### PR DESCRIPTION
**Pull request by sakshamarora1**
Added support for landscape view in the following simulations:

  - Epicycloid (Pencil of Lines)
  - Epicycloid Curve
  - Lissajous Curve
  - Maurer Rose
  - Rose Pattern

### CODE REVIEW

**Comment by yashlamba** [comment]
These are the major problems I have noticed (Flutter for Web):
1. Rose Pattern returns the following error when in landscape and denominator = 0
`Another exception was thrown: RangeError (index): Index out of range: no indices are valid: 0`
image
> The spacing between the sliders also needs to be adjusted for all simulations, as evident above.

2. Toothpick Pattern has an overflow error in landscape mode, might get fixed by some minor adjustments.

3. Lissajous Throws the same error as Rose Pattern once animate is true. It might have issues with sleep function but the UI also disfigures.

4. Animate is not working with many simulations, maybe we can get rid of the sleep issue in this PR only.

File: lib/src/simulations/epicycloid.dart [hunk 1]
```diff
@@ -24,28 +24,6 @@ class _NormalEpicycloidCurveState extends State<NormalEpicycloidCurve> {
...
-    ys.clear();
-    time = 0;
-    SystemChrome.setPreferredOrientations([
-      DeviceOrientation.portraitUp,
-      DeviceOrientation.portraitDown,
-      DeviceOrientation.landscapeLeft,
-      DeviceOrientation.landscapeRight,
-    ]);
-    super.dispose();
-  }
```
Comment (yashlamba) [inline lib/src/simulations/epicycloid.dart]: We can't just get rid of these methods, just remove the orientation stuff.

**Review by yashlamba — COMMENTED** [review]
I'll review further ASAP.

**Review by yashugarg — CHANGES_REQUESTED** [review]
- [x] Remove SizedBox() from List view when in landscape mode from all simulations.
- [x] In Epicycloid Pattern (Pencil of Lines), "Points" overlaps the curve.
Screenshot_2020-10-26-00-24-01-88_6722ee9149bcc9e95a5626b4285cc503 1

File: lib/src/simulations/toothpick.dart [hunk 2]
```diff
@@ -112,32 +112,20 @@ class _ToothpickPatternState extends State<ToothpickPattern> {
...
   }
 
   @override
   void dispose() {
-    SystemChrome.setPreferredOrientations([
-      DeviceOrientation.portraitUp,
-      DeviceOrientation.portraitDown,
-      DeviceOrientation.landscapeLeft,
-      DeviceOrientation.landscapeRight,
-    ]);
```
Comment (yashlamba) [inline lib/src/simulations/toothpick.dart]: I think we should keep this for toothpick pattern?

**Review by yashlamba — CHANGES_REQUESTED** [review]
Getting rid of the sized boxes is necessary, they don't add to the UI. I like how everything works now though, I didn't have any other immediate issues on Pixel 2 XL. Great work!

File: lib/src/simulations/toothpick.dart (hunk 2 above)
Comment (sakshamarora1) [reply lib/src/simulations/toothpick.dart]: Yeah, my bad, I'll remove that.

**Comment by yashlamba** [comment]
Fix for the range error/distorted UI: 
https://github.com/cod-ed/simulate/blob/9524b04a5617bda2a1423b642424ed325fc08484/lib/src/simulations/rose_pattern.dart#L367
to
```dart
if (points.length > 0) {
      canvas.drawPoints(PointMode.polygon, points, paint);
}
```

**Review by yashlamba — APPROVED** [review]

**Review by yashugarg — APPROVED** [review]

**Pull request merged**